doc/developer/cmakeutilities.md

```diff
@@ -0,0 +1,309 @@
+ 
+  - `<TARGET>`. The name of the target to configure
+  - `<DIRECTORY>`. The directory in the source tree that contains the resource to install.
+  - `<BUILD_LOCATION>`. If given, `<DIRECTORY>` will be linked in the build tree at this location. If not, the resource directory will be linked to `${CMAKE_CURRENT_BINARY_DIR}/../Resources` directory.
+  - `<PREFIX>`. If given, the resources will be installed into the directory `<install_prefix>/Resources/<PREFIX>/`. If not, the resources will be installed into the directory `<install_prefix>/Resources/` where `<install_prefix>`is the installation directory of the target.
+  - `<FILES>`. If given, this allow to install only the given files from `<DIRECTORY>`. If not, all the files from `<DIRECTORY>` will be installed.
+ 
+For a Radium-based software component, resources can be of several type :
+  - Text files, that contains some glsl source code for used shaders.
+  - Images used internally as textures or source of data.
+  - Precomputed data, stored in whatever format.
+  
+To access resources, the component might rely on a Ra::Core::Resources::ResourcesLocator object. Such an object allows to access files located in some predefine directories
+  - Radium default resource directory. This directory is located at `<prefix>/Resources` where `<prefix>` is the Radium installation directory.
+  - Current executable directory. This directory is where the currently running executable is located.
+  
+In order to allow client component to access their own resources, and to allow that either on the build-tree or once the component is installed without having to recompile or modify anything, this function both link (on supported systems, copy elsewhere) and install the component resources at the given places.
+
+As the process must be system and component independent it is the responsibility of the client package to configure the proper paths for resources management.
+
+According to the installation layout of the Radium lib, that matches the GNU standard, but also to the MacOs bundle architecture, Resources might be localized relatively to any binary object (library, executable, plugin, ...)
+The Radium installation hierarchy, that embed base libraries, their resources and some executable file is the following.
+```
+<prefix>/
+├── include/
+|   ├── Core/
+|   ├── Engine/
+|   ...
+├── lib/
+|   ├── cmake/
+|   |   └── Radium/
+|   ├──  libCore.so
+|   ...
+├── bin/
+|   ├──  main-app
+|   ...
+├── Resources/
+|   ├── Shaders/
+|   ├── Config/
+|   ...
+```
+In this hierarchy, Resources are located in the relative path `../Resources/` from any binary (executable or lib).
+The source code structure of the Radium libraries allow to have the same property on the buildtree.
+When a Radium component need to access some resources, it could either use the Ra::Core::Resources::ResourcesLocator::getRadiumResourcesDir() method or search for the resource relatively to a symbol defined in the component.
```

Comment:
```suggestion
The source code structure of the Radium libraries allows to have the same property on the buildtree.
When a Radium component needs to access some resources, it could either use the Ra::Core::Resources::ResourcesLocator::getRadiumResourcesDir() method or search for the resource relatively to a symbol defined in the component.
```